File: src/gestorAplicacion/carcel/Antidelito.java
/*
 * La clase Antidelito representa las posibles acciones o trabajos que prisionero puede realizar para
 * disminuir su tiempo de condena. Se relaciona directamente con la clase Prisionero.
 * 
 * Posee los siguientes atributos:
 * - codigo (int): Permite la identificacion del Antidelito por medio de un entero.
 * - nombre (String): Permite la identificacion del Antidelito por medio de un String.
 * - descripcion (String): Contiene la descripcion de la accion o trabajo que el Prisionero puede realizar.
 * - antidelitos (Hashtable<Integer,Antidelito>): Permite llevar un registro de los Antidelitos creados, donde
 *   la clave de la Hashtable es el codigo del Antidelito y Antidelito es this.Antidelito.
 * - rebajaCondena (long): Representa el tiempo en meses que disminuye el Antidelito al tiempo de condena del 
 *   Prisionero que la realiza.
 */

package gestorAplicacion.carcel;
import java.io.Serializable;
import java.util.Hashtable;

public class Antidelito implements Serializable{
   
	private static final long serialVersionUID = 1L;
	
	private static Hashtable<Integer,Antidelito> antidelitos= new Hashtable<Integer,Antidelito>();
	private int codigo;
    private String nombre;
    private String descripcion;
    private long rebajaCondena; /*meses*/
    
    /*
     * Cuando se crea un Antidelito, se agrega a una lista general de antidelitos.
     */
    public Antidelito(int codigo, String nombre, String descripcion, long rebajaCondena) {
    	this.codigo = codigo;
    	this.nombre = nombre;
    	this.descripcion = descripcion;
    	this.rebajaCondena = rebajaCondena;
    	antidelitos.put(this.codigo, this);
    }

	public static Hashtable<Integer, Antidelito> getAntidelitos() {return antidelitos;}
	public static void setAntidelitos(Hashtable<Integer, Antidelito> antidelitos) {Antidelito.antidelitos = antidelitos;}
	
	public int getCodigo() {return codigo;}
	public void setCodigo(int codigo) {this.codigo = codigo;}

	public String getNombre() {return nombre;}
	public void setNombre(String nombre) {this.nombre = nombre;}

	public long getRebajaCondena() {return rebajaCondena;}
	public void setRebajaCondena(long rebajaCondena) {this.rebajaCondena = rebajaCondena;}
    
	@Override
	public String toString() {
		
		return "Codigo: " + this.getCodigo() + "\n" 
				+ "Nombre: " + this.getNombre() + "\n"
				+ "Rebaja condena: " + this.getRebajaCondena() + " meses" + "\n" + "\n";
	}

	public String getDescripcion() {return descripcion;}
	public void setDescripcion(String descripcion) {this.descripcion = descripcion;}
}

File: Python/src/gestorAplicacion/antidelito.py
class Antidelito:
    _antidelitos = {}

    def __init__(self, nombre: str, descripcion: str, rebajaCondena: int):
        self._codigo = len(Antidelito._antdelitos) + 1
        self._nombre = nombre
        self._descripcion = descripcion
        self._rebajaCondena = rebajaCondena
        Antidelito._antidelitos[self._codigo] = self

    def getCodigo(self):
        return self._codigo
    def setCodigo(self, codigo: int):
        self._codigo = codigo

    def getNombre(self):
        return self._nombre
    def setNombre(self, nombre: str):
        self._nombre = nombre

    def getDescripcion(self):
        return self._descripcion
    def setDescripcion(self, descripcion: str):
        self._descripcion = descripcion

    def getRebajaCondena(self):
        return self._rebajaCondena
    def setRebajaCondena(self, rebajaCondena: int):
        self._rebajaCondena = rebajaCondena

    @classmethod
    def getAntidelitos(cls):
        return cls._antidelitos
    @classmethod
    def setAntidelitos(cls, antidelitos):
        cls._antidelitos = antidelitos

    def __str__(self) -> str: #Decidir implementacion
        pass

    def __str__(self) -> str:
        return ("Delito: "+ str(self._codigo) + "\n"  
                + "Nombre: " + self.getNombre() + "\n"
                + "Rebaja condena: " + str(self.getRebajaCondena()) + " meses" +"\n"
                + "Descripcion: " + self.getDescripcion() + "\n")        